src/geolocation/identifier.rs

```diff
@@ -0,0 +1,40 @@
+use geoip2_city::CityApiResponse;
+use std::fmt;
+use std::fmt::{Display, Formatter};
+
+#[derive(Clone, Debug, Hash, Ord, PartialOrd, Eq, PartialEq)]
+pub struct DatacenterIdentifier {
+    autonomous_system_number: u32,
+    country_code: String,
+    city_name: Option<String>,
+}
+
+impl From<CityApiResponse> for DatacenterIdentifier {
+    fn from(val: CityApiResponse) -> Self {
+        Self {
+            autonomous_system_number: val.traits.autonomous_system_number,
+            country_code: val
+                .country
+                .map(|c| c.iso_code)
+                .unwrap_or_else(|| "XX".to_string()),
```

Comment:
Maybe "??" just to completely disambiguate with valid country codes?

Reply:
[The code XX is reserved to indicate an unknown state/entity/organisation in ISO-3166](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2#XX)

Reply:
I see. OK then.